docs/architecture/ADR-003-borrow-assets.md

```diff
@@ -5,106 +5,114 @@
 
-The calculated borrow limit, which weighs collateral uTokens against borrowed assets (e.g. `u/uatom:ether`) is derived from combining the two above.
+The calculated borrow limit, which weighs collateral uTokens against borrowed assets (e.g. `u/atom:ether`) is derived from combining the two above. The weight of each uToken is defined as `CollateralWeight`.
 
 Note also that as a consequence of uToken interest, the asset value of uToken collateral increases over time, meaning a user who repays positions in full and redeems collateral uTokens will receive back more base assets than they deposited originally, reducing the effective interest.
 
 ## Detailed Design
 
-For the purposes of borrowing and repaying assets, as well as marking uTokens as collateral, the `umee/x/leverage` module does not mint or burn tokens. It uses its keeper to remember open positions and user collateral settings, and the `x/bank` module to perform all necessary balance checks and token transfers.
+For the purposes of borrowing and repaying assets, as well as marking uTokens as collateral, the `umee/x/leverage` module does not mint or burn tokens. It stores borrower open positions and collateral settings, and the `x/bank` module to perform all necessary balance checks and token transfers. User collateral (uTokens) are deposited in `x/leverage` module and withdrawn back to the user `x/bank` account balance when the user disables uTokens as a collateral.
 
-To prevent users from transferring uTokens when doing so would lower their borrow limit below their current total borrowed amount, uTokens that are enabled as collateral are stored in the `x/leverage` module account, and moved back to the user's wallet when they are disabled as collateral.
-As long as it would not reduce a user's borrow limit below their borrowed value, `MsgWithdrawAsset` can withdraw base assets in exchange uTokens used as collateral, even when they are stored in the module account.
+During every operation which involves borrow position or collateral we check that the _borrow limit_ invariant holds.
 
-Because borrow limits weight the value of different token denominations together, the calculation will require the use of `x/oracle`.
+`x/oracle` module is used to provide exchange rates of tokens to calculate borrow limit.
 
-### Basic Message Types
+### API
 
-To implement the borrow/repay functionality of the Asset Facility, the three common message types will be:
+To implement the borrow/repay functionality of the Asset Facility, the three message types are defined:
 
 ```go
-// MsgSetCollateral - a borrower enables or disables a specific utoken type in their wallet to be used as collateral
+// MsgSetCollateral - a borrower enables or disables a specific uToken type in their wallet to be used as collateral
 type MsgSetCollateral struct {
   Borrower sdk.AccAddress `json:"borrower" yaml:"borrower"`
+  // not a uToken
```

Comment:
Currently a uToken - although it might be cleaner to simplify to base asset before 3.0.0

Reply:
I wanted to address Asset - Token nomenclature in other PR.

Reply:
Clarification: This message uses `uToken`, not ` token` denomination - so it would succeed on input `u/uumee` and fail on input `uumee`.